bin/runtime/src/lib.rs

```diff
@@ -655,21 +742,23 @@ impl_runtime_apis! {
     }
 
     impl primitives::AlephSessionApi<Block> for Runtime {
-        fn next_session_authorities() -> Result<Vec<AlephId>, AlephApiError> {
-            Aleph::next_session_authorities()
-        }
-
         fn authorities() -> Vec<AlephId> {
             Aleph::authorities()
         }
 
-        fn session_period() -> u32 {
-            Aleph::session_period()
+        fn millisecs_per_block() -> u64 {
+            Elections::millisecs_per_block()
         }
 
-        fn millisecs_per_block() -> u64 {
-            Aleph::millisecs_per_block()
+        fn session_period() -> u32 {
+            Elections::session_period()
         }
 
+        fn next_session_authorities() -> Result<Vec<AlephId>, AlephApiError> {
+                Session::queued_keys()
+                .iter()
+                .map(|(_, key)| key.get(AlephId::ID).ok_or(AlephApiError::DecodeKey))
+                .collect::<Result<Vec<AlephId>, AlephApiError>>()
```

Comment:
Why did we move the implementation from pallet Aleph to here?

Reply:
The original reason was that the main functionality is provided by pallet session and all the types are from primitives, so there is no dependence on pallet aleph. Explicitly, ofc, since this works because pallet aleph implements `OneSessionHanlder`.

Reply:
Ok, fine, but just to clarify -- why do you think there is a connection between this working, and Aleph implementing `OneSessionHandler`. I think this works just because there is a dedicated Aleph key (with id `AlephId::ID`) in `SessionKeys`

Reply:
Don't remember now but I've seen some convoluted macro magic that allows keys in opaque_keys macro only if this trait is implemented. I'll try to find it.

Reply:
Yeah, I saw it once and I don't with anyone to look at it again :P I'm resolving :)